src/cmds/pbsnodes.c

```diff
@@ -244,110 +237,63 @@ encode_to_json(struct batch_status *bstat) {
 					ivalue = (long int)value;
 					if (strcmp(pc, "") == 0) {
 						if (value == ivalue)
-							node = add_json_node(JSON_VALUE, JSON_INT, next->resource, &ivalue);
+							if (add_json_node(JSON_VALUE, JSON_INT, next->resource, &ivalue) == NULL)
+								return 1;
 						else
-							node = add_json_node(JSON_VALUE, JSON_FLOAT, next->resource, &value);
-						if (node == NULL)
-							return 1;
+							if (add_json_node(JSON_VALUE, JSON_FLOAT, next->resource, &value) == NULL)
+								return 1;
 					} else if (str[0] != '0') {
-						node = add_json_node(JSON_VALUE, JSON_STRING, next->resource, str);
-						if (node == NULL)
+						if (add_json_node(JSON_VALUE, JSON_STRING, next->resource, str) == NULL)
 							return 1;
 					}
 				} else {
 					if (str[0] != '0') {
-						node = add_json_node(JSON_VALUE, JSON_STRING, next->resource, str);
-						if (node == NULL)
+						if (add_json_node(JSON_VALUE, JSON_STRING, next->resource, str) == NULL)
 							return 1;
 					}
 				}
-				if (next->next) {
-					if (strcmp(next->next->name, "resources_assigned") == 0)
-						next = next->next;
-					else {
-						/* In case of an empty object */
-						if (node->node_type == JSON_OBJECT) {
-							node = add_json_node(JSON_OBJECT_END, JSON_NULL, NULL, NULL);
-							if (node == NULL)
-								return 1;
-						} else {
-							node->node_type = JSON_OBJECT_END;
-							pattr = next;
-							next = NULL;
-							break;
-						}
-					}
+				if (next->next == NULL || strcmp(next->next->name, "resources_assigned")) {
+					/* Nothing left in resources_assigned, close object */
+					if (add_json_node(JSON_OBJECT_END, JSON_NULL, NULL, NULL) == NULL)
+						return 1;
+					pattr = next;
+					next = NULL;
 				} else {
-					/* In case of an empty object adding an extra node to end the object. */
-					if (node->node_type == JSON_OBJECT) {
-						node = add_json_node(JSON_OBJECT_END, JSON_NULL, NULL, NULL);
-						if (node == NULL)
-							return 1;
-					} else {
-						node->node_type = JSON_OBJECT_END;
-						pattr = next;
-						next = NULL;
-						break;
-					}
+					next = next->next;
 				}
 			}
-		} else {
```

Comment:
What happened to this else case?

Reply:
The previous tree was:
```
>if resources_available
>else if resources_assigned
>else
>>if jobs
>>else
>>>extras
```
I changed it to

```
>if resources_available
>else if resources_assigned
>else if jobs
>else
>>extras
```

Reply:
The else was moved to line 290. On 291, the node gets created and `pbs_json::add_json_node` figures out the type.